modules/service/abci.go

```diff
@@ -1,45 +1,41 @@
 package service
 
 import (
-	"github.com/irisnet/irishub/app/v1/auth"
-	"github.com/irisnet/irishub/app/v1/service/tags"
-	"github.com/irisnet/irishub/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/irisnet/irishub/modules/service/internal/types"
 )
 
-func EndBlocker(ctx types.Context, keeper Keeper) (resTags types.Tags) {
-	ctx = ctx.WithLogger(ctx.Logger().With("handler", "endBlock").With("module", "iris/service"))
-	logger := ctx.Logger()
+// EndBlocker handles block ending logic
+func EndBlocker(ctx sdk.Context, keeper Keeper) {
+	logger := keeper.Logger(ctx)
+
 	// Reset the intra-transaction counter.
 	keeper.SetIntraTxCounter(ctx, 0)
 
-	resTags = types.NewTags()
-	params := keeper.GetParamSet(ctx)
+	params := keeper.GetParams(ctx)
 	slashFraction := params.SlashFraction
 
 	activeIterator := keeper.ActiveRequestQueueIterator(ctx, ctx.BlockHeight())
 	defer activeIterator.Close()
+
 	for ; activeIterator.Valid(); activeIterator.Next() {
 		var req SvcRequest
-		keeper.cdc.MustUnmarshalBinaryLengthPrefixed(activeIterator.Value(), &req)
+		keeper.GetCdc().MustUnmarshalBinaryLengthPrefixed(activeIterator.Value(), &req)
```

Comment:
can ues `ModuleCdc `

Reply:
I think we should only use ModuleCdc in msg.ValidateBasic